### ISSUE TITLE
Uncaught TypeError: Blockly.createSvgElement is not a function

### ISSUE THREAD

**Issue by rwytrwal** [HendrikD/blockly-plugins#2]
Uncaught TypeError: Blockly.createSvgElement is not a function
    at Blockly.RenderedConnection.Blockly.Connection.typeHighlight (http://localhost:63342/google/blockly/blockly-type-indicator/typeIndicator.js:178:24)
    at http://localhost:63342/google/blockly/blockly-type-indicator/typeIndicator.js:113:76
    at Blockly.ConnectionDB.forEach (native)
    at Blockly.BlockSvg.newOnMouseMove_ [as onMouseMove_] (http://localhost:63342/google/blockly/blockly-type-indicator/typeIndicator.js:106:26)
    at HTMLDocument.wrapFunc (http://localhost:63342/google/blockly/core/blockly.js:452:14)

I did all steps as you wrote to install it but I get that kind of error. Any ideas ?

**Comment by rwytrwal** [comment]
I need to change : 
`var allBlocks = draggingWorkspace.getAllBlocks();` 
to
`var allBlocks = Blockly.mainWorkspace.getAllBlocks();`

and 
`return Blockly.createSvgElement('path', {`
to
`return Blockly.utils.createSvgElement('path', {`
and its work well :)

**Comment by HendrikD** [comment]
Hi,

thanks for your feedback!

The blockly developers change some of the functions used by the type indicator from time to time. I then need to change some bits to make it work again like you did now.  I think there could be a problem with your fix for the draggingWorkspace part: It won't work if you have multiple workspaces.
Although you fixed this in your own code base, I need to fix it in the repository to solve this issue.

Greetings,
Hendrik.

**Comment by rachel-fenichel** [comment]
Hi @HendrikD!  

I just did a major rewrite of Blockly's dragging code to beat down some technical debt.  Unfortunately I broke your code :/

The good news is that all of the dragging code is now in one place, instead of scattered between block, field, and workspace mouse handlers.  
- gesture.js handles events.  It decides what kind of gesture is happening (click, block drag, workspace drag, etc) and routes events accordingly
- block_dragger.js handles the work of dragging a block or stack of blocks.  It has a constructor, `startBlockDrag`, `dragBlock`, and `endBlockDrag`.
- dragged_connection_manager.js updates connections during drags, and is responsible for finding the closest eligible connection and highlighting or unhighlighting it as needed during a drag.  You probably want to override its `update` and `applyConnections` functions.  `update` runs for every mouse move during a block drag.  `applyConnections` is called when a block drag is ending, and it connects the block to the closest connection (if any) and cleans up unneeded highlighting.

I think you'll want to move [this code](https://github.com/SPE-Systemhaus/blockly-type-indicator/blob/master/typeIndicator.js#L37) into/overriding `DraggedConnectionManager.prototype.applyConnections`.

[This code](https://github.com/SPE-Systemhaus/blockly-type-indicator/blob/master/typeIndicator.js#L79) would go into/override the `DraggedConnectionManager` constructor.

If you have questions, let me know and I'll see if I can help.  I'm kicking around ideas for a more formal API for things like this, but it's not yet fully formed.  I'll be keeping your use case in mind.

**Comment by HendrikD** [comment]
Hi @rachel-fenichel ,

thank you for your clarification. The changes you are talking about are in blocklys develop branch right? I fixed the problem with the master branch for now but i will look into the develop branch soon.

I'am glad that you are looking for better and more stable methods for people to customize blockly to their needs using code. There are a lot of great projects using blockly but it's not easy to share improvements you made in your own projects with the blockly community. I think if blockly would provide some support for developing easy shareable"plugins" blockly could be taken to a whole new level.

Greetings,
Hendrik

tl;dr: Great idea!

**Comment by rachel-fenichel** [comment]
Hi @HendrikD,

The changes I was talking about are on develop, but I'm doing a release to master in the next week.

Yes, plugins would be awesome, as long as we found the right way to make and use them.

Cheers,
Rachel
